src/sentry/processing/backpressure/redis.py

```diff
@@ -12,44 +12,23 @@
 Cluster = Union[RedisCluster, Redis]
 
 
-class RedisMemoryUsageMetrics:
-    """
-    This class allows querying for the memory usage percentage of a number of
-    redis clusters, each consisting of a number of nodes.
-    """
-
-    def __init__(self, clusters: Sequence[Cluster]) -> None:
-        self.clusters = clusters
-
-    def query_usage_percentage(self) -> float:
-        """
-        Queries the memory usage (using the `INFO` command) of all the cluster nodes in the
-        given clusters, and returns the *highest* usage percentage of any participating node.
-        """
-
-        highest_usage = 0.0
-
-        infos = (info for cluster in self.clusters for info in iter_cluster_node_infos(cluster))
-        for info in infos:
-            # or alternatively: `used_memory_rss`?
-            memory_used = info.get("used_memory", 0)
-            # `maxmemory` might be 0 in development
-            memory_available = info.get("maxmemory", 0) or info.get("total_system_memory", 0)
-
-            if memory_available:
-                node_usage = min(memory_used / memory_available, 1.0)
-                highest_usage = max(highest_usage, node_usage)
+def get_memory_usage(info: Mapping[str, Any]) -> Tuple[int, int]:
```

Comment:
_nit_: Consider using a named tuple in the return value or even return `ServiceMemory` right away? To avoid circular imports, you could move these utility types to a `base` module.

Same applies to `query_rabbitmq_memory_usage`